src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -1581,252 +1581,381 @@ bool AletheProofPostprocessCallback::update(Node res,
+    // c)).
     //
-    // ------------------------------------------------------ LA_DISEQUALITY
-    //  (VP1: (cl (or (= x c) (not (<= x c)) (not (<= c x)))))
-    // -------------------------------------------------------- OR
-    //  (VP2: (cl (= x c) (not (<= x c)) (not (<= c x))))
+    // The convertion into Alethe is based on la_disequality, which has much
+    // the same semantics as ARITH_TRICHOTOMY. The following subproof is
+    // common to all the cases (we will refer to it as PI_0):
     //
-    // If C = (> x c) or C = (< x c) post-processing has to be added. In these
-    // cases resolution on VP2 A B yields (not (<=x c)) or (not (<= c x)) and
-    // comp_simplify is used to transform it into C. Otherwise,
+    // ------------------------------------------------------ la_disequality
+    //  (cl (or (= x c) (not (<= x c)) (not (<= c x))))
+    // -------------------------------------------------------- or
+    //  (cl (= x c) (not (<= x c)) (not (<= c x)))
     //
-    //  VP2   A   B
-    // ---------------- RESOLUTION
-    //  (cl C)*
-    //
-    // * the corresponding proof node is C
+    // The transformations also use the COMP_SIMPLIFY rule in Alethe, which
+    // connects strict and non-strict inequalities. The details for each
+    // conversion are given for each case.
     case ProofRule::ARITH_TRICHOTOMY:
     {
       bool success = true;
       Node equal, lesser, greater;
-
       Kind k = res.getKind();
-      if (k == Kind::EQUAL)
-      {
-        equal = res;
-        if (children[0].getKind() == Kind::LEQ)
-        {
-          greater = children[0];
-          lesser = children[1];
-        }
-        else
-        {
-          greater = children[1];
-          lesser = children[0];
-        }
-      }
-      // Add case where res is not =
-      else if (res.getKind() == Kind::GT)
+      Assert(k == Kind::EQUAL || k == Kind::GT || k == Kind::LT)
+          << "kind is " << k << "\n";
+      Node x = res[0], c = res[1];
+      switch (k)
       {
-        greater = res;
-        if (children[0].getKind() == Kind::NOT)
+        case Kind::EQUAL:
         {
-          equal = children[0];
-          lesser = children[1];
+          Trace("alethe-proof") << "..case EQUAL\n";
+          Node leq, geq;
+          if (children[0].getKind() == Kind::LEQ)
+          {
+            leq = children[0];
+            geq = children[1];
+          }
+          else
+          {
+            leq = children[1];
+            geq = children[0];
+          }
+          Node leqInverted = nm->mkNode(Kind::LEQ, geq[1], geq[0]);
+          // The subproof built is (where @p1 is the premise for "geq", @p2 is
+          // "leqInverted")
+          //
+          // PI_1:
+          //   with @p0: (= (=> x c) (<= c x))
+          //   with @p1: (=> x c)
+          //   with @p2: (<= c x)
+          //
+          // ----- comp_simp  -------------------equiv_pos2   --- geq
```

Comment:
Should be comp_simplify

Reply:
I was trying to save space, but I fixed and managed to fit in the 80chars now.